src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/Formatting/FormattingHandler.ts

```diff
@@ -5,27 +5,38 @@
 
 import * as vscode from 'vscode';
 import { RequestType } from 'vscode-languageclient';
+import { IRazorDocument } from '../Document/IRazorDocument';
 import { RazorDocumentManager } from '../Document/RazorDocumentManager';
+import { RazorDocumentSynchronizer } from '../Document/RazorDocumentSynchronizer';
 import { RazorLanguageServerClient } from '../RazorLanguageServerClient';
 import { RazorLogger } from '../RazorLogger';
 import { convertTextEditToSerializable, SerializableTextEdit } from '../RPC/SerializableTextEdit';
 import { SerializableFormattingParams } from './SerializableFormattingParams';
 import { SerializableFormattingResponse } from './SerializableFormattingResponse';
+import { SerializableOnTypeFormattingParams } from './SerializableOnTypeFormattingParams';
 
 export class FormattingHandler {
-    private static readonly provideFormattingEndpoint = 'textDocument/formatting';
+    private static readonly provideFormattingEndpoint = 'razor/htmlFormatting';
+    private static readonly provideOnTypeFormattingEndpoint = 'razor/htmlOnTypeFormatting';
     private formattingRequestType: RequestType<SerializableFormattingParams, SerializableFormattingResponse, any> = new RequestType(FormattingHandler.provideFormattingEndpoint);
+    private onTypeFormattingRequestType: RequestType<SerializableOnTypeFormattingParams, SerializableFormattingResponse, any> = new RequestType(FormattingHandler.provideOnTypeFormattingEndpoint);
     private emptyFormattingResponse = new SerializableFormattingResponse();
 
     constructor(
         private readonly documentManager: RazorDocumentManager,
+        private readonly documentSynchronizer: RazorDocumentSynchronizer,
         private readonly serverClient: RazorLanguageServerClient,
         private readonly logger: RazorLogger) { }
 
     public register() {
-        return this.serverClient.onRequestWithParams<SerializableFormattingParams, SerializableFormattingResponse, any>(
+        // tslint:disable-next-line: no-floating-promises
```

Comment:
nit: we previously had a work item that got rid of these disabling lines (https://github.com/dotnet/razor/issues/6908), would be nice to not re-introduce them if possible